ui/components/app/confirm-title/confirm-title.js

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { TransactionType } from '../../../../shared/constants/transaction';
+import { PRIMARY } from '../../../helpers/constants/common';
+import { Text } from '../../component-library';
+import {
+  FONT_WEIGHT,
+  TextVariant,
+} from '../../../helpers/constants/design-system';
+import UserPreferencedCurrencyDisplay from '../user-preferenced-currency-display';
+
+const ConfirmTitle = ({ title, hexTransactionAmount, txData }) => {
+  const isContractInteraction =
+    txData.type === TransactionType.contractInteraction;
+
+  const hideTitle =
+    (isContractInteraction || txData.type === TransactionType.deployContract) &&
+    txData.txParams?.value === '0x0';
+
+  if (hideTitle) {
+    return null;
+  }
+
+  if (title) {
+    return (
+      <Text
+        as="h3"
+        ellipsis
+        title={title}
+        variant={
+          title.length < 10 ? TextVariant.headingLg : TextVariant.headingMd
+        }
+        fontWeight={FONT_WEIGHT.NORMAL}
+      >
+        {title}
+      </Text>
+    );
+  }
+
+  return (
+    <Text
+      as="h3"
+      ellipsis
+      fontWeight={FONT_WEIGHT.NORMAL}
+      variant={TextVariant.headingMd}
+    >
+      <UserPreferencedCurrencyDisplay
+        ethLogoHeight={24}
+        hideLabel={!isContractInteraction}
+        showCurrencySuffix={isContractInteraction}
+        showEthLogo
+        type={PRIMARY}
+        value={hexTransactionAmount}
+      />
+    </Text>
```

Comment:
nice cleanup! 

does this code also support removing the use of `titleComponent` passed from `ui/pages/confirm-transaction-base/confirm-transaction-base.component.js` to ConfirmPageContainer?